### PR TITLE
Form Bug Fix and Redirection

### PR DESCRIPTION
**Pull request by edbella**
#### What does this PR achieve?
Form Feedback and Redirect on successful sign in or sign up.

### CODE REVIEW

File: src/components/signin/components/AntdForm.jsx [hunk 1]
```diff
@@ -112,11 +110,25 @@ function formItem(dispatch, handleSubmit) {
...
     const dispatch = useDispatch();
     const data = useSelector(state => state);
-    const { signIn: { email, password } } = data;
+    const {
+        signIn: {
+            email, password, isLoggingIn, isLoggingOut, isAuthenticated, loginError, logoutError,
+        },
+    } = data;
     const user = { email, password };
+    if (isLoggingIn === true || isLoggingOut === true) {
```
Comment (justiceotuya) [inline src/components/signin/components/AntdForm.jsx]: no need of using the ```isLoggingIn === true```   you can just do ```if (isLoggingIn || isLoggingOut ) {...}```

Comment (justiceotuya) [reply src/components/signin/components/AntdForm.jsx]: same as others

```
if (isLoggingIn || isLoggingOut) {
   message.loading({ content: 'Request processing' }, 5);
    } else if (isAuthenticated) {
        message.success({ content: 'Signed In successfully' }, 15);
    } else if (loginError && loginError !== ' ' ) {
        message.error({ content: loginError }, 15);
    } else if (logoutError && logoutError !== ' '  ) {
        message.error({ content: logoutError }, 15);
    } else message.destroy();
```

File: src/components/signup/components/FormCard.jsx [hunk 2]
```diff
@@ -137,11 +137,11 @@ export function FormCard() {
         },
     } = data;
```
Comment (justiceotuya) [inline src/components/signup/components/FormCard.jsx]: do same here

File: src/components/navbar/components/UserName.jsx [hunk 3]
```diff
@@ -1,25 +1,49 @@
...
+ * @function
+ * @param {function} dispatch - useDispatch hook
+ */
+const menu = dispatch => (
+    <Menu onClick={() => { dispatch(requestLogout()); }}>
+        {NAVBAR_MENU_STRINGS.map(item => {
+            const { key, text } = item;
+            return (
+                <Menu.Item key={key}>
+                    <span>{text}</span>
```
Comment (enyeInc) [inline src/components/navbar/components/UserName.jsx]: no need for the `span` here

File: src/components/navbar/components/UserName.jsx [hunk 4]
```diff
@@ -1,25 +1,49 @@
...
+import { useDispatch } from 'react-redux';
+import { WELCOME_TEXT, NAVBAR_MENU_STRINGS } from '../constants';
+import { requestLogout } from '../../signin/actions';
+
+/**
+ * Renders the Menu items in the dropdown menu
+ * @function
+ * @param {function} dispatch - useDispatch hook
+ */
+const menu = dispatch => (
```
Comment (enyeInc) [inline src/components/navbar/components/UserName.jsx]: do not pass dispatch as a prop here, just call is directly in this component

File: src/components/navbar/components/UserName.jsx [hunk 5]
```diff
@@ -1,25 +1,49 @@
...
+                </Menu.Item>
+            );
+        })}
+    </Menu>
+);
 
 export default function UserName(props) {
     const { user } = props;
     const { displayName } = user;
+    const dispatch = useDispatch();
```
Comment (enyeInc) [inline src/components/navbar/components/UserName.jsx]: see previous comment

File: src/components/navbar/components/UserName.jsx [hunk 6]
```diff
@@ -1,25 +1,49 @@
...
     return (
-        <div>
-            <p className="m-0">
+        <Dropdown className="m-0" overlay={menu(dispatch)} trigger={['click']}>
```
Comment (enyeInc) [inline src/components/navbar/components/UserName.jsx]: be explicit with your `className`s, "m-0" meaning nothing

File: src/components/signin/components/SignIn.jsx [hunk 7]
```diff
@@ -1,17 +1,30 @@
...
 import FormView from './FormView';
 import PageHelmet from './PageHelmet';
 
+function redirectOnSuccess(isAuthenticated) {
+    return (
+        <>
+            {isAuthenticated ? (<Redirect push to="/dashboard" />) : false}
+        </>
+    );
+}
```
Comment (enyeInc) [inline src/components/signin/components/SignIn.jsx]: this should simply be
```
return isAuthenticated ? <Redirect push to="/dashboard" /> : false;
```

File: src/components/signin/components/SignIn.jsx [hunk 8]
```diff
@@ -1,17 +1,30 @@
...
  */
 
 export function SignIn() {
+    const data = useSelector(state => state);
+    const { signIn: { isAuthenticated } } = data;
     return (
         <>
             <PageHelmet />
             <FormView />
+            {redirectOnSuccess(isAuthenticated)}
```
Comment (enyeInc) [inline src/components/signin/components/SignIn.jsx]: line 27 should just be
```
{isAuthenticated ? (<Redirect push to="/dashboard" />) : false}
```

File: src/components/signup/components/SignupForm.jsx [hunk 9]
```diff
@@ -1,26 +1,62 @@
...
+            <Row type="flex" justify="center" className="hv-100">
+                <Col span={21} className="h-100">
+                    <Row type="flex" className="h-100" justify="center" align="middle">
+                        <Col md={22} lg={14} className="form-holder" data-aos="fade">
+                            {renderSignUpViews}
+                            {redirectOnSuccess(isAuthenticated, success)}
+                        </Col>
+                    </Row>
+                </Col>
+            </Row>
```
Comment (enyeInc) [inline src/components/signup/components/SignupForm.jsx]: do not do this nested `row/col` pattern.

Comment (enyeInc) [reply src/components/signup/components/SignupForm.jsx]: Use [CSS-grid](https://css-tricks.com/snippets/css/complete-guide-grid/) instead

File: src/components/navbar/components/UserName.jsx (hunk 4 above)
Comment (edbella) [reply src/components/navbar/components/UserName.jsx]: I've tried that but I kept getting the error `React Hook "useDispatch" is called in function "menu" which is neither a React function component or a custom React Hook function`
@enyeInc

File: src/components/signup/components/SignupForm.jsx (hunk 9 above)
Comment (edbella) [reply src/components/signup/components/SignupForm.jsx]: Unfortunately, that's @Segun-Ade 's code. But I'll see how I can rework it in this PR.

**Pull request merged**